include/openthread/platform/border_routing.h

```diff
@@ -65,6 +65,18 @@ extern "C" {
  */
 extern void otPlatBorderRoutingProcessIcmp6Ra(otInstance *aInstance, const uint8_t *aMessage, uint16_t aLength);
 
+/**
+ * Used to notify the platforms whether they should request an prefix via DHCPv6 PD for Thread interface.
+ * When aShouldRequest is false, the platform must release the prefix lease.
+ *
+ * Requires `OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE`.
+ *
+ * @param[in] aInstance  A pointer to an OpenThread instance.
+ * @param[in] aShouldRequest  Indicate if the platform need to request/release the prefix.
+ *
+ */
+extern void otPlatBorderRoutingShouldRequestDhcp6Pd(otInstance *aInstance, bool aShouldRequest);
```

Comment:
This API seems to be a normal platform API so should not have `extern`.

- A normal API is implemented by platform layer and OT stack will call it
- Use of `extern` in platform API means this is callback from platform to OT stack, i.e. OT stack implements and provides this API for the platform layer will call it

Reply:
This is an API for other teams to reference. Ideally there could be another APP/Process calling this API to get the state, or being notified, and then control the platform to request/release prefix.

Reply:
Should remove the `extern` in the declaration.

Here is the policy:
- An `otPlat` API implemented by platform layer for use by OT stack should not be defined as `extern` (e.g. `otPlatRadioTransmit`).
- An `otPlat` API implemented by OT stack for use by platform layer should be defined as `extern` (e.g., `otPlatRadioTxDone`). These are often callbacks.

Reply:
> This is an API for other teams to reference. Ideally there could be another APP/Process calling this API to get the state, or being notified, and then control the platform to request/release prefix.

No, I guess the app will **implement** this API to get the state.

For querying the state, the app should call `otBorderRoutingDhcp6PdGetState`.

Reply:
Maybe I should rename the API to processStateChange()? My understanding is that different platforms will implement different methods to handle the prefix request/release